survey/survey_app/views.py

```diff
@@ -28,6 +38,9 @@ class AnswerViewSet(viewsets.ModelViewSet):
     queryset = Answer.objects.all()
     serializer_class = AnswerSerializer
 
+    def perform_create(self, serializer):
```

Comment:
Oftenly we do not need to override in view layer @newDaria . You can just add current user as creator in serializer layer.

Reply:
```
serializers.py

User = get_user_model()

def get_request_user(serializer):
    try:
        return serializer.context['request'].user
    except KeyError:
        return None

class SurveySerializer(serializers.ModelSerializer):
    questions = QuestionSerializer(many=True, read_only=True)
    creator = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())  # or StringRelatedField

    class Meta:
        model = Survey
        fields = ['id', 'title', 'questions', 'creator']

    def create(self, validated_data):
        # Set the creator field to the current user
        request_user = get_request_user(self)
        if request_user is not None:
            validated_data['creator'] = request_user
        return super().create(validated_data)
```